### ISSUE TITLE
`chunk` method doesn't work

### ISSUE THREAD

**Issue by nikurt** [near/near-jsonrpc-client-rs#24]
This code returns an error for both testnet and mainnet
```
            let chunk_request = RpcChunkRequest {
                chunk_reference: ChunkReference::ChunkHash {
                    chunk_id: CryptoHash::from_str("6ccdyqFVuKCNrYqzMFCCmCppXvSvBwNc7JirP8By4GQM").unwrap(),
                },
            };
            let chunk = JsonRpcClient::connect("https://archival-rpc.testnet.near.org")
                .call(chunk_request)
                .await
                .map_err(|e| simple_error!("{:#?}", e))?;
            log::debug!("Got chunk {}: {:#?}", chunk_hash, chunk);
```

The error is
```
ServerError(
    RequestValidationError(
        ParseError {
            error_message: \"Failed parsing args: data did not match any variant of untagged enum ChunkReference\",
        },
    ),
)"
```

The RPC node is aware of that chunk, this httpie request returns a huge response:
`http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=chunk params:='{ "chunk_id": "6ccdyqFVuKCNrYqzMFCCmCppXvSvBwNc7JirP8By4GQM"}'`

**Comment by miraclx** [comment]
I'm taking a look.

**Comment by miraclx** [comment]
Fixed on master with f40ad743653ad2a1a9eb5eaa96c302c9b531bf40

Took another look at https://github.com/near/nearcore/blob/28c10371f2304cfc7097f88e7dbbcafb18873b8e/chain/jsonrpc-primitives/src/types/chunks.rs#L72, turns out the deserialization logic doesn't expect a `ChunkRefererence` contained in an array.

Thanks for spotting this! Expect it to land in the next crates.io release this weekend.

In the meantime, you can resolve this by patching the dependency in your manifest to use this revision.

```toml
[patch.crates-io.near-jsonrpc-client]
git = "https://github.com/near/near-jsonrpc-client-rs"
rev = "f40ad743653ad2a1a9eb5eaa96c302c9b531bf40"
```
